pkg/strategy/dca2/strategy.go

```diff
@@ -43,16 +45,32 @@ type Strategy struct {
 	// OrderGroupID is the group ID used for the strategy instance for canceling orders
 	OrderGroupID uint32 `json:"orderGroupID"`
 
+	// RecoverWhenStart option is used for recovering dca states
+	RecoverWhenStart bool `json:"recoverWhenStart"`
+
+	// KeepOrdersWhenShutdown option is used for keeping the grid orders when shutting down bbgo
+	KeepOrdersWhenShutdown bool `json:"keepOrdersWhenShutdown"`
+
 	// log
 	logger    *logrus.Entry
 	LogFields logrus.Fields `json:"logFields"`
 
+	// PrometheusLabels will be used as the base prometheus labels
+	PrometheusLabels prometheus.Labels `json:"prometheusLabels"`
+
 	// private field
 	mu                   sync.Mutex
 	takeProfitPrice      fixedpoint.Value
 	startTimeOfNextRound time.Time
 	nextStateC           chan State
 	state                State
+
+	// callbacks
+	readyCallbacks    []func()
+	positionCallbacks []func(*types.Position)
+	profitCallbacks   []func(*types.ProfitStats)
+	closedCallbacks   []func()
+	errorCallbacks    []func(error)
```

Comment:
can we share the common callbacks with grid2?

Reply:
I think we can't..... because the ProfitStats.

Reply:
ok, but maybe for readyCallback, closeCallbacks and errorCallbacks?